packages/runtime/runtime-definitions/src/id-compressor/idCompressor.ts

```diff
@@ -0,0 +1,140 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	SessionSpaceCompressedId,
+	OpSpaceCompressedId,
+	SessionId,
+	FinalCompressedId,
+	StableId,
+} from "./identifiers";
+import {
+	IdCreationRange,
+	SerializedIdCompressor,
+	SerializedIdCompressorWithNoSession,
+	SerializedIdCompressorWithOngoingSession,
+} from "./persisted-types";
+
+/**
+ * This is the set of functions on an IdCompressor that a DDS author shouldn't be concerned with.
+ * Operations here are automatically handled at the SharedObject level.
+ */
+export interface IIdCompressorCore {
+	/**
+	 * Finalizes the supplied range of IDs (which may be from either a remote or local session).
+	 * @param range - the range of session-local IDs to finalize.
+	 */
+	finalizeCreationRange(range: IdCreationRange): void;
+
+	/**
+	 * Returns a range of local IDs created by this session in a format for sending to the server for finalizing.
+	 * The range will include all local IDs generated via calls to `generateCompressedId` since the last time this method was called.
+	 * @returns the range of session-local IDs, which may be empty. This range must be sent to the server for ordering before
+	 * it is finalized. Ranges must be sent to the server in the order that they are taken via calls to this method.
+	 */
+	takeNextCreationRange(): IdCreationRange;
+
+	/**
+	 * Returns an iterable of all IDs created by this compressor.
+	 */
+	getAllIdsFromLocalSession(): IterableIterator<SessionSpaceCompressedId>;
```

Comment:
I'm actually pretty sure this method (and implementation) aren't needed anymore. They were used for the phantom compressor stashed ops implementation, which we've decided to move away from. Is it used anywhere else?

Reply:
Removed in latest commit along with its tests.